persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java

```diff
@@ -93,6 +93,24 @@ public static BuildProject getBuildProject(Path projectPath) throws BalException
         }
         return buildProject;
     }
+
+    public static BuildProject getFullBuildProject(Path projectPath) throws BalException {
+        BuildProject buildProject = BuildProject.load(projectPath.toAbsolutePath());
+        Package currentPackage = buildProject.currentPackage();
+        PackageCompilation compilation = currentPackage.getCompilation();
+        DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+        if (diagnosticResult.hasErrors()) {
+            StringBuilder errorMessage = new StringBuilder();
+            errorMessage.append("Error occurred when validating the project. ");
+            for (Diagnostic diagnostic : diagnosticResult.errors()) {
+                errorMessage.append(System.lineSeparator());
+                errorMessage.append(diagnostic);
+
+            }
+            throw new BalException(errorMessage.toString());
+        }
+        return buildProject;
+    }
```

Comment:
Shall we merge this with the `getBuildProject` function and add new input param `skipSourcePath`. like below,

```
public static BuildProject getBuildProject(Path projectPath, bollean skipGeneratedDir) throws BalException {
    if (skipGeneratedDir && !diagnostic.location().lineRange().filePath().startsWith(
                        GENERATED_SOURCE_DIRECTORY)) {
       ....
    }
}
```

Reply:
updated in [c5fe5ed](https://github.com/ballerina-platform/persist-tools/pull/87/commits/c5fe5ed207ffd718257a70831ff1c4493909e6e6)